core/src/main/java/bisq/core/xmr/jsonrpc/MoneroWalletRpc.java

```diff
@@ -0,0 +1,169 @@
+package bisq.core.xmr.jsonrpc;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import bisq.core.xmr.jsonrpc.result.Address;
+import bisq.core.xmr.jsonrpc.result.Balance;
+import bisq.core.xmr.jsonrpc.result.MoneroTransfer;
+import bisq.core.xmr.jsonrpc.result.MoneroTransferList;
+import bisq.core.xmr.jsonrpc.result.MoneroTx;
+
+public class MoneroWalletRpc {
+	
+	protected Logger log = LoggerFactory.getLogger(MoneroWalletRpc.class);
+	
+	private MoneroRpcConnection rpcConnection; 
+
+	public MoneroWalletRpc(MoneroRpcConnection rpcConnection) {
+		this.rpcConnection = rpcConnection;
+	}
+	
+	public String getPrimaryAddress() {
+		Map<String, Object> params = new HashMap<>();
+		params.put("account_index", 0);
+		Map<String, Object> response = rpcConnection.sendJsonRequest("get_address", params);
+		log.debug("response => {}", response);
+		
+		Address address = JsonUtils.DEFAULT_MAPPER.fromJson(JsonUtils.DEFAULT_MAPPER.toJson(response.get("result")), Address.class);
```

Comment:
The utility methods within JsonUtils.java (`serialize()`, `deserialize()`, `toMap()`, and `fromMap()`) should be used for JSON serialization/deserialization instead of calling GSON_MAPPER `toJson()` or `fromJson()` directly which is implementation-specific and inconsistent with other usage.

Reply:
This is resolved.